lib/util/http/index.ts

```diff
@@ -283,114 +288,99 @@ export class Http<Opts extends HttpOptions = HttpOptions> {
   }
 
   getJson<T>(url: string, options?: Opts): Promise<HttpResponse<T>>;
-  getJson<T>(
+  getJson<T, Schema extends ZodType<T> = ZodType<T>>(
     url: string,
-    schema: ZodSchema<T>
-  ): Promise<HttpResponse<Infer<typeof schema>>>;
-  getJson<T>(
+    schema: Schema
+  ): Promise<HttpResponse<Infer<Schema>>>;
+  getJson<T, Schema extends ZodType<T> = ZodType<T>>(
     url: string,
     options: Opts,
-    schema: ZodSchema<T>
-  ): Promise<HttpResponse<Infer<typeof schema>>>;
-  getJson<T = unknown>(
+    schema: Schema
+  ): Promise<HttpResponse<Infer<Schema>>>;
+  getJson<T = unknown, Schema extends ZodType<T> = ZodType<T>>(
     arg1: string,
-    arg2?: Opts | ZodSchema,
-    arg3?: ZodSchema
+    arg2?: Opts | Schema,
+    arg3?: Schema
   ): Promise<HttpResponse<T>> {
     const args = this.resolveArgs(arg1, arg2, arg3);
     return this.requestJson<T>('get', args);
   }
 
-  headJson<T>(url: string, options?: Opts): Promise<HttpResponse<T>>;
-  headJson<T>(
-    url: string,
-    schema: ZodSchema<T>
-  ): Promise<HttpResponse<Infer<typeof schema>>>;
-  headJson<T>(
-    url: string,
-    options: Opts,
-    schema: ZodSchema<T>
-  ): Promise<HttpResponse<Infer<typeof schema>>>;
-  headJson<T = unknown>(
-    arg1: string,
-    arg2?: Opts | ZodSchema,
-    arg3?: ZodSchema
-  ): Promise<HttpResponse<T>> {
-    const args = this.resolveArgs(arg1, arg2, arg3);
-    return this.requestJson<T>('head', args);
+  headJson(url: string, httpOptions?: Opts): Promise<HttpResponse<undefined>> {
```

Comment:
we should remove this in a future PR, it makes no sense to have this. we should use normal head function instead as it never returns json

Reply:
I don't know how to express this in types. We still need `HttpResponse` because we may be interested in response headers, however maybe `never` instead of `undefined` is the better option?

Reply:
Hah, fair enough